### PR TITLE
bpo-46658: shutil Lib enabling sendfile on solaris systems as it supports

### PR DESCRIPTION
**Pull request by devnexen**
copy between file descriptors.

https://bugs.python.org/issue46658

### CODE REVIEW

File: Lib/shutil.py [hunk 1]
```diff
@@ -42,7 +42,8 @@
 COPY_BUFSIZE = 1024 * 1024 if _WINDOWS else 64 * 1024
 # This should never be removed, see rationale in:
 # https://bugs.python.org/issue43743#msg393429
-_USE_CP_SENDFILE = hasattr(os, "sendfile") and sys.platform.startswith("linux")
+# senfile on solaris systems is capable of copying on regular file descriptors.
+_USE_CP_SENDFILE = hasattr(os, "sendfile") and (sys.platform.startswith("linux") or sys.platform.startswith("sunos"))
```
Comment (JelleZijlstra) [inline Lib/shutil.py]:
```suggestion
_USE_CP_SENDFILE = hasattr(os, "sendfile") and sys.platform.startswith(("linux", "sunos"))
```

File: Misc/NEWS.d/next/Library/2022-02-06-15-53-30.bpo-46658.oRp0C-.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Enabling ``os.sendfile`` on shlib on Solaris based systems.
```
Comment (JelleZijlstra) [inline Misc/NEWS.d/next/Library/2022-02-06-15-53-30.bpo-46658.oRp0C-.rst]:
```suggestion
Enable use of :func:`os.sendfile` in :mod:`shutil` on Solaris.
```

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
How should this be tested?

From looking at test_shutil, it seems like it should run sendfile-based tests on Solaris automatically, but we don't run Solaris in normal CI. Perhaps we need to trigger the buildbots.

**Comment by encukou** [comment]
Closing in favour of #23893.

**Pull request closed without merging**